src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -299,6 +307,60 @@ public CommandBase driveToNearest() {
 
     }
 
+    public CommandBase driveUntilTipped() {
+        return cmd().onExecute(() -> {
+            move(0.0, BALANCE_SPEED, 0.0);
+
+        }).runsUntil(() -> Math.abs(this.getTilt()) > 9);
+    }
+
+    public double getTilt() {
+        return this.map.gyro().getRotation2d().getCos()
+                * Units.radiansToDegrees(this.map.gyro().getRotation3d().getY())
+                + this.map.gyro().getRotation2d().getSin()
+                        * Units.radiansToDegrees(this.map.gyro().getRotation3d().getX());
+    }
+
+    public CommandBase balance() {
+
+        PersistenceCheck balancedCheck = new PersistenceCheck(50,
+                () -> Math.abs(getTilt()) < 6);
+        return cmd().onExecute(() -> {
+            Rotation3d rotationVelocity = this.map.gyro().getRotationalVelocity();
+
+            double velocityThresholdDegreesPerSec = 4.0;
```

Comment:
This should be a constant (`private final double`) defined at the top of the class.